app/scripts/angular-apimock.js

```diff
@@ -234,7 +260,8 @@ angular.module('apiMock', [])
 
 			// replace apiPath with mockDataPath.
 			var oldPath = req.url;
-			var redirectedPath = req.url.replace(config.apiPath, config.mockDataPath);
+
+			var redirectedPath = req.url.replace(apiPathMatched(req.url, config.apiPath), config.mockDataPath);
```

Comment:
I was thinking about once I wrote the test cases, and agree with you that would be possible to get collisions at some point, it was enough for me and I didn't have too much time to get a proper solucion, probably would be a breaking change to get a one to one map between apiPath and mockData routes.